packages/worker/src/internals/worker-pool/BasicPool.ts

```diff
@@ -114,15 +114,15 @@ export class BasicPool<TSuccess, TPayload> implements IWorkerPool<TSuccess, TPay
       worker,
       isAvailable,
       isFaulty,
-      register: (predicateId, payload, onSuccess, onFailure) => {
+      register: (predicateId, payload, state, onSuccess, onFailure) => {
         if (!isAvailable()) {
           throw new Error('This instance of PooledWorker is currently in use');
         }
         const currentRunId = ++runIdInWorker;
         registration = { currentRunId, onSuccess, onFailure };
         const message: PoolToWorkerMessage<TPayload> = {
           targetPredicateId: predicateId,
-          payload,
+          content: state !== undefined ? { source: 'worker', ...state } : { source: 'main', payload },
```

Comment:
Probably worth taking the payload from the outside as an argument of the `register` function

Reply:
Should we keep the pool agnostic of the details within the content? If so we could just take a content:TContent and let our callers know what's going inside